### PR TITLE
v6: rename IAPD option to be shorter; add Options getters

### PR DESCRIPTION
**Pull request by hugelgupf**
Also removes superfluous GetOne/Del proxy functions and introduces PDOptions and PrefixOptions.

Signed-off-by: Chris Koch <chrisko@google.com>

### CODE REVIEW

**Comment by hugelgupf** [comment]
updated ptal

**Comment by pmazzini** [comment]
There is a merge conflict.

**Comment by hugelgupf** [comment]
Will rebase brb

File: dhcpv6/option_iaprefix.go [hunk 1]
```diff
@@ -7,16 +7,35 @@ import (
...
 //
 // This module defines the OptIAPrefix structure.
 // https://www.ietf.org/rfc/rfc3633.txt
 type OptIAPrefix struct {
 	PreferredLifetime uint32
 	ValidLifetime     uint32
-	prefixLength      byte
-	ipv6Prefix        net.IP
-	Options           Options
+	Prefix            *net.IPNet
```
[the remark this answers is not in the document]
Comment (hugelgupf) [reply dhcpv6/option_iaprefix.go]: i think nil is no different than the zero value, and net.IPNet in the stdlib is always used with a pointer, so i don't want to deviate (even the Stringer has the pointer)

File: dhcpv6/option_iapd.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	"github.com/u-root/u-root/pkg/uio"
+)
+
+// PDOptions are options used with the IAPD (prefix delegation) option.
+//
+// RFC 3633 describes that IA_PD-options may contain the IAPrefix option and
+// the StatusCode option.
+//
+// RFC 8415 Appendix C does not list the Status Code option as valid, but it
```
[the remark this answers is not in the document]
Comment (hugelgupf) [reply dhcpv6/option_iapd.go]: good catch. thanks.

File: dhcpv6/option_iapd.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+// does say that the previous text in RFC 8415 Section 21.21 supersedes that
+// table. Section 21.21 does mention the Status Code option.
+type PDOptions struct {
+	Options
+}
+
+// Prefixes are the prefixes associated with this delegation.
+func (po PDOptions) Prefixes() []*OptIAPrefix {
+	opts := po.Options.Get(OptionIAPrefix)
+	var pre []*OptIAPrefix
```
[the remark this answers is not in the document]
Comment (hugelgupf) [reply dhcpv6/option_iapd.go]: done

**Comment by hugelgupf** [comment]
ptal

**Comment by hugelgupf** [comment]
woops. other stuff has changed. brb.

**Comment by hugelgupf** [comment]
ptal. only rtr7 failing. will have a PR for that later.

File: dhcpv6/option_iaprefix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Natolumin) [reply dhcpv6/option_iaprefix.go]: Good point, and the nil pointer footgun issue is solved by #365 as well

**Comment by hugelgupf** [comment]
Yeah, rtr7 at least. But they've got modules now.

**Pull request merged**